### PR TITLE
Improve isometry docs

### LINKED ISSUE #14308: Add more docs on Isometry types
A bit more docs could be nice. Maybe briefly something on:

- What is an isometry?
- Why and how is this useful?
- A small code example (creating an isometry and applying a transformation or two with it)

I'm not sure how mathematical we want to be, but I think it could be good to briefly mention that these isometries specifically represent (I think) rigid motion / Euclidean motion (see [Wikipedia](https://en.wikipedia.org/wiki/Rigid_transformation)), and are essentially transformations that preserve Euclidean distance (shape and size remains the same), excluding reflections. More formally, these handedness-preserving rigid motions belong to the Special Euclidean group SE(n) where n denotes the dimension, but I'm not sure if we want to be *that* technical here.

_Originally posted by @Jondolf in https://github.com/bevyengine/bevy/pull/14269#discussion_r1672809364_

### PR DESCRIPTION
**Pull request by Jondolf**
# Objective

Fixes #14308.

#14269 added the `Isometry2d` and `Isometry3d` types, but they don't have usage examples or much documentation on what the types actually represent or what they may be useful for.

In addition, their module is public and the types are not re-exported at the crate root, unlike all the other core math types like Glam's types, direction types, and `Rot2`.

## Solution

Improve the documentation of `Isometry2d` and `Isometry3d`, explaining what they represent and can be useful for, along with doc examples on common high-level usage. I also made the way the types are exported consistent with other core math types.

This does add some duplication, but I personally think having good docs for this is valuable, and people are also less likely to look at the module-level docs than type-level docs.

### CODE REVIEW

File: crates/bevy_math/src/isometry.rs [hunk 1]
```diff
@@ -206,7 +268,81 @@ impl UlpsEq for Isometry2d {
     }
 }
 
-/// An isometry in three dimensions.
+/// An isometry in three dimensions, representing a rotation followed by a translation.
+/// This can often be useful for expressing relative positions and transformations from one position to another.
+///
+/// In particular, this type represents a distance-preserving transformation known as a *rigid motion* or a *direct motion*,
+/// and belongs to the *Special Euclidean group* SE(3). This includes translation and rotation, but excludes reflection.
```
Comment (alice-i-cecile) [inline crates/bevy_math/src/isometry.rs:275]: Wikipedia or similar links would be nice for further reading.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by torsteingrindvik** [comment]
As a Bevy user I'm still wondering: Why would I use the isometry APIs when the use of transform is pervasive, and it seems I can get to the same end results using that.
If that could be added to the docs I think it would clear up the role of isometries quite a bit.

File: crates/bevy_math/src/isometry.rs (hunk 1 above)
Comment (mweatherley) [reply crates/bevy_math/src/isometry.rs:275]: Seconding this!

File: crates/bevy_math/src/lib.rs [hunk 2]
```diff
@@ -33,6 +33,7 @@ pub use aspect_ratio::AspectRatio;
 pub use common_traits::*;
 pub use direction::*;
 pub use float_ord::*;
+pub use isometry::{Isometry2d, Isometry3d};
```
Comment (mweatherley) [inline crates/bevy_math/src/lib.rs:36]: This is a good change; I'm glad you thought of this.

**Review by mweatherley — APPROVED** [review]

File: crates/bevy_math/src/lib.rs [hunk 3]
```diff
@@ -19,7 +19,7 @@ mod compass;
 pub mod cubic_splines;
 mod direction;
 mod float_ord;
-pub mod isometry;
+mod isometry;
```
Comment (mnmaita) [inline crates/bevy_math/src/lib.rs:22]: Maybe this is super nitpicky, but I feel like this makes the PR a breaking change, right? If so, could you please detail that Isometry2d and Isometry3d are now imported directly from `bevy_math` in the Migration Guide section of the PR description? Great documentation addition btw!

Comment (Jondolf) [reply crates/bevy_math/src/lib.rs:22]: The isometry types are completely new types that aren't in any Bevy release yet, so I don't think this counts as a breaking change even though it is "breaking" relative to #14269 which introduced the types

**Pull request merged**